internal/app/machined/pkg/controllers/cluster/local_affiliate.go

```diff
@@ -197,6 +197,8 @@ func (ctrl *LocalAffiliateController) Run(ctx context.Context, r controller.Runt
 						}
 
 						spec.KubeSpan.Endpoints = endpoints
+					} else {
+						spec.KubeSpan = cluster.KubeSpanAffiliateSpec{}
```

Comment:
I'm not sure I follow this thing, but can we not just do a `if spec.KubeSpan == nil { spec.KubeSpan = cluster.KubeSpanAffiliateSpec{} }` _before_ this `if` and thus eliminate the `else`?

Reply:
#readability

Reply:
This one is simple - Affiliate contains current state for the local node. If KubeSpan was enabled, Affiliate might have had KubeSpan information. But on this reconcile loop we see that KubeSpan is disabled, so we need to clear KubeSpan settings of the Affiliate.

We don't need to clear it in general

Reply:
but I got your point, I'll update it, thanks